optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -210,32 +213,95 @@ private void applySolverProperties(RecorderContext recorderContext,
-                    findImplementingClass(DotNames.EASY_SCORE_CALCULATOR, indexView));
-            scoreDirectorFactoryConfig.setConstraintProviderClass(
-                    findImplementingClass(DotNames.CONSTRAINT_PROVIDER, indexView));
-            scoreDirectorFactoryConfig.setIncrementalScoreCalculatorClass(
-                    findImplementingClass(DotNames.INCREMENTAL_SCORE_CALCULATOR, indexView));
-            ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
-            if (classLoader.getResource(SolverBuildTimeConfig.DEFAULT_SCORE_DRL_URL) != null) {
-                scoreDirectorFactoryConfig.setScoreDrlList(Collections.singletonList(
-                        SolverBuildTimeConfig.DEFAULT_SCORE_DRL_URL));
+            ScoreDirectorFactoryConfig scoreDirectorFactoryConfig =
+                    defaultScoreDirectoryFactoryConfig(indexView, effectiveConstraintsDrl);
+            solverConfig.setScoreDirectorFactoryConfig(scoreDirectorFactoryConfig);
+        } else {
+            ScoreDirectorFactoryConfig scoreDirectorFactoryConfig = solverConfig.getScoreDirectorFactoryConfig();
+            if (constraintsDrlFromProperty.isPresent()) {
+                scoreDirectorFactoryConfig.setScoreDrlList(Collections.singletonList(constraintsDrlFromProperty.get()));
+            } else {
+                if (scoreDirectorFactoryConfig.getScoreDrlList() == null) {
+                    defaultConstraintsDrl.ifPresent(resolvedConstraintsDrl -> scoreDirectorFactoryConfig
+                            .setScoreDrlList(Collections.singletonList(resolvedConstraintsDrl)));
+                }
             }
-            if (scoreDirectorFactoryConfig.getEasyScoreCalculatorClass() == null
-                    && scoreDirectorFactoryConfig.getConstraintProviderClass() == null
-                    && scoreDirectorFactoryConfig.getIncrementalScoreCalculatorClass() == null
-                    && scoreDirectorFactoryConfig.getScoreDrlList() == null) {
-                throw new IllegalStateException("No classes found that implement "
-                        + EasyScoreCalculator.class.getSimpleName() + ", "
-                        + ConstraintProvider.class.getSimpleName() + " or "
-                        + IncrementalScoreCalculator.class.getSimpleName() + ", nor a "
-                        + SolverBuildTimeConfig.DEFAULT_SCORE_DRL_URL + " resource.");
+        }
+
+        if (solverConfig.getScoreDirectorFactoryConfig().getScoreDrlList() != null) {
+            boolean isDroolsDynamicPresent = isClassDefined("org.drools.dynamic.DynamicServiceRegistrySupplier");
+            boolean isDroolsStaticPresent = isClassDefined("org.drools.statics.StaticServiceRegistrySupplier");
+
+            if (!isDroolsDynamicPresent) {
+                throw new IllegalStateException(
+                        "Using scoreDRL in Quarkus, but the dependency drools-core-dynamic is not on the classpath.\n"
```

Comment:
grammar: "Using scoreDRL in Quarkus, but"
```suggestion
                        "If scoreDRL is used in Quarkus, the dependency drools-core-dynamic must be on the classpath.\n"
```